addons/binding/org.openhab.binding.hs110/src/main/java/org/openhab/binding/hs110/handler/HS110Handler.java

```diff
@@ -0,0 +1,225 @@
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.binding.ConfigStatusThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.RefreshType;
+import org.eclipse.smarthome.core.types.State;
+import org.eclipse.smarthome.core.types.UnDefType;
+import org.openhab.binding.hs110.internal.HS110;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link HS110Handler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Christian Fischer - Initial contribution
+ */
+public class HS110Handler extends ConfigStatusThingHandler {
+
+    private Logger logger = LoggerFactory.getLogger(HS110Handler.class);
+
+    private HS110 plug;
+    private static final int REFRESH_DEFAULT = 5;
+    private BigDecimal refresh;
+    private ScheduledFuture<?> refreshJob;
+
+    private String sysinfoData;
+    private String energyData;
+
+    public HS110Handler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        if (command instanceof OnOffType) {
+            if (channelUID.getId().equals(CHANNEL_SWITCH)) {
+                logger.debug("Switching {} {}", thing.getUID().getAsString(), command.toFullString());
+                try {
+                    plug.sendSwitch((OnOffType) command);
+                } catch (IOException e) {
+                    logger.error("Failed sending command {} to {}", command.toFullString(),
+                            thing.getUID().getAsString(), e);
+                }
+            }
+        }
+
+        if (command instanceof RefreshType) {
+            boolean success = updateData();
+            if (success) {
+                switch (channelUID.getId()) {
+                    case CHANNEL_WATTAGE:
+                        updateState(channelUID, getWattage());
+                        break;
+                    case CHANNEL_TOTAL:
+                        updateState(channelUID, getTotal());
+                        break;
+                    case CHANNEL_SWITCH:
+                        updateState(channelUID, getState());
+                        break;
+                    case CHANNEL_SYSINFO:
+                        updateState(channelUID, getSysinfo());
+                        break;
+                    default:
+                        logger.debug("Command received for an unknown channel: {}", channelUID.getId());
+                        break;
+                }
+            }
+        } else {
+            logger.debug("Command {} is not supported for channel: {}", command, channelUID.getId());
+        }
+
+    }
+
+    @Override
+    public void dispose() {
+        refreshJob.cancel(true);
```

Comment:
There are some possibilities  that the `dispose` is called before `initialize` is called/completed? Better do a `null`-check on `refreshJob` and null it out after you cancelled it.